syntax = "proto3";
package permission;





/*
ValidateArtifactPermission请求
*/
message ValidateArtifactPermissionRequest {
    /*
    需要鉴权的用户名
    */
    string user = 1 ;
    /*
    需要鉴权的权限点
    */
    string action = 2 ;
    /*
    如果一个权限点有数据权限,但是不想校验数据权限,这个参数传true，默认为false
    */
    bool validate_action_only = 3 ;
    /*
    包id
    */
    string package_id = 4 ;
    /*
    版本id, 多个用,连接，并把batch参数置为true
    */
    string version_id = 5 ;
    /*
    是否批量鉴权版本
    */
    bool batch = 6 ;
}

/*
ValidateArtifactPermission返回
*/
message ValidateArtifactPermissionResponse {
message Data {
    /*
    是否鉴权成功
    */
    bool accepted = 1 ;
}
        
    
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    结果信息
    */
    string msg = 2 ;
    /*
    返回数据
    */
    Data data = 3 ;
}

/*
ValidateArtifactPermissionApi返回
*/
message ValidateArtifactPermissionResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    permission.ValidateArtifactPermissionResponse data = 4 ;
}

